pkg/server/server.go

```diff
@@ -357,6 +376,24 @@ func (s *ProxyServer) removeFrontendsForBackendConn(agentID string, backend Back
 	return ret, nil
 }
 
+func (s *ProxyServer) removeFrontendsForStream(uuid string) []*ProxyClientConnection {
+	var ret []*ProxyClientConnection
+	s.fmu.Lock()
+	defer s.fmu.Unlock()
+	for agentID, frontends := range s.frontends {
+		for connID, frontend := range frontends {
```

Comment:
I think we're a bit inconsistent with what `frontend` refers to in the code. Maybe not here, but at some point we should probably audit the existing usages and come up with less ambiguous terms.

Reply:
Agreed. When I added `GrpcFrontend` I had in mind future renaming. Maybe:

`ProxyClientConnection` -> `ProxyConnection`
`frontends` -> `proxyConnections`

Reply:
+1. I know we already discussed this but I think we should document some of the terms and then audit to make sure that our usage matches our definitions.

For me:

- Frontend -- The client (KAS) from the Konn servers perspective
- Backend -- The agent from the Konn servers perspective
- Tunnel -- The "outer" TCP/IP connection. Generally equivalent to the GRPC stream (or http-connect). We generally have 2 tunnel segments. One from client to konn server. Another from konn server to konn agent.
- Connection -- Need a better name. The "inner"  TCP/IP connection. The connectioned request from the client to the destination.

Reply:
I will take this documentation (and rename) to a followup.